From 84170efe6c1dfb1d9fd3da2c9d05f52bdf03aa1f Mon Sep 17 00:00:00 2001
From: Brano Zarnovican <branislav.zarnovican@gooddata.com>
Date: Sat, 18 May 2013 13:16:39 +0200
Subject: [PATCH] FIX: switch back to SNAT rule to any target if

without this hack, instance cannot access itself by the public IP

with "if device"
  124 17334 SNAT       all  --  *      bond0.279  172.30.48.33         0.0.0.0/0           to:166.78.83.223
without
 3355  212K SNAT       all  --  *      *       172.30.48.33         0.0.0.0/0           to:166.78.83.223
---
 nova/network/linux_net.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/nova/network/linux_net.py b/nova/network/linux_net.py
index 600d5e5..9994ef2 100644
--- a/nova/network/linux_net.py
+++ b/nova/network/linux_net.py
@@ -638,8 +638,6 @@ def remove_floating_forward(floating_ip, fixed_ip, device):
 
 def floating_forward_rules(floating_ip, fixed_ip, device):
     rule = '-s %s -j SNAT --to %s' % (fixed_ip, floating_ip)
-    if device:
-        rule += ' -o %s' % device
     return [('PREROUTING', '-d %s -j DNAT --to %s' % (floating_ip, fixed_ip)),
             ('OUTPUT', '-d %s -j DNAT --to %s' % (floating_ip, fixed_ip)),
             ('float-snat', rule)]
-- 
1.8.1.4

